python-package/xgboost/dask.py

```diff
@@ -907,6 +905,13 @@ def _filter_empty(
     raise ValueError("None of the workers can provide a valid result.")
 
 
+def _check_workers_are_alive(workers: List[str], client: "distributed.Client") -> None:
```

Comment:
Not entirely sure what replacing wait with check does on some clusters. The wait was there for some cluster types like GKE, where the worker can take some time to be in sync with the scheduler somehow and `client.scheduler_info()["workers"].` was not particularly reliable.

Reply:
@trivialfis: You raise a good point here. When using an async client, `client.scheduler_info()` would get refreshed by a periodic callback. So this _could_ lead to a situation where `client.scheduler_info()` is outdated. I've updated the PR to request the data directly from the scheduler, ensuring that `workers` is up-to-date. Generally, if a worker is alive and has executed work (e.g., loaded partitions of the `DaskDMatrix`), the scheduler will be aware of the worker.